tensorboard/webapp/core/views/hash_storage_test.ts

```diff
@@ -64,14 +65,58 @@ describe('hash storage test', () => {
     getPluginIdSpy = spyOn(deepLinker, 'getPluginId');
   });
 
-  it('sets the hash to plugin id by replacing on first load', () => {
-    store.overrideSelector(getActivePlugin, 'foo');
+  it('sets hash to plugin id when plugin id changes from null to value', () => {
+    const setPluginIdCalls: Array<{
+      id: string | null;
+      option: SetStringOption;
+    }> = [];
+    setPluginIdSpy.and.callFake(
+      (id: string | null, option: SetStringOption) => {
+        setPluginIdCalls.push({
+          id,
+          option,
+        });
+      }
+    );
+    store.overrideSelector(getActivePlugin, null);
     const fixture = TestBed.createComponent(HashStorageContainer);
     fixture.detectChanges();
-    expect(setPluginIdSpy).toHaveBeenCalledWith('foo', {
-      useLocationReplace: true,
-      defaultValue: '',
-    });
+
+    store.overrideSelector(getActivePlugin, null);
+    store.refreshState();
+    fixture.detectChanges();
+
+    store.overrideSelector(getActivePlugin, 'foo');
+    store.refreshState();
+    fixture.detectChanges();
+
+    store.overrideSelector(getActivePlugin, null);
+    store.refreshState();
+    fixture.detectChanges();
+
+    expect(setPluginIdCalls).toEqual([
+      {
+        id: '',
+        option: {
+          useLocationReplace: false,
+          defaultValue: '',
+        },
+      },
+      {
+        id: 'foo',
+        option: {
+          useLocationReplace: true,
+          defaultValue: '',
+        },
+      },
+      {
+        id: '',
+        option: {
+          useLocationReplace: false,
+          defaultValue: '',
+        },
+      },
+    ]);
   });
 
   it('sets the hash to empty string when activePlugin is not set', () => {
```

Comment:
Just curious, can you remind me what the actual scenarios are for the follow two tests (where we no longer will use replaceState)?

My current understanding is that having an unset activePlugin would indicate a case where there are no active plugins at all (and it should show the no-dashboards-active message).  So in that case it would seem like there shouldn't be a navigation at all, and I assume pushState() itself would ignore the new state if it's the same URL, right?  So that seems fine.

However, I'm not sure what it means for activePlugin to be the empty string?

Reply:
> I assume pushState() itself would ignore the new state if it's the same URL, right? So that seems fine.

That's not strictly correct but it is true because we do `window.location.hash = <newPluginId>` and `hash = ""` is no-op when there is no change.

`activePlugin = null`, activePlugin is not set/bootstrapped yet
`activePlugin = ""`, user explicitly used "localhost:6006/#" 

Upon more careful reflection, I think it is proper to use the replaceState on the first render of the hash_storage_component so I kept the old behavior and modified the test accordingly.